### PR TITLE
Create separate dns record resources

### PR DESCRIPTION
**Pull request by samcrang**
I still need to `push-config`s and generate the address config in the discovery environment. I'll do this when/if we this gets merged.

### CODE REVIEW

**Comment by samcrang** [comment]
There's also so tidying up we can do (specifically around our `.tfvars`) that can happen after we deploy.

File: ansible/group_vars/tag_Environment_discovery [hunk 1]
```diff
@@ -20,8 +20,10 @@ register_groups:
     - datatype
     - field
 
-  multi:
+  address:
     - address
```
Comment (philandstuff) [inline ansible/group_vars/tag_Environment_discovery]: wait, why are you doing this? `address` only needs to be separate in `test`, right?

File: aws/modules/register/variables.tf [hunk 2]
```diff
@@ -1,74 +1,9 @@
...
-}
-
-variable "dns_zone_id" {
-  description = "Route 53 zone id for DNS records for ELBs (ie public records)"
-}
-
-variable "dns_ttl" {
-  default = "300"
-  description = "TTL, in seconds, for Route 53 DNS records"
+variable name {}
```
Comment (philandstuff) [inline aws/modules/register/variables.tf]: why did you remove the `description`s?

**Comment by philandstuff** [comment]
there's a typo in `availiabilty_checks.tf ` filename 😆

File: ansible/group_vars/tag_Environment_discovery (hunk 1 above)
Comment (philandstuff) [inline ansible/group_vars/tag_Environment_discovery]: ah I see you explained in bc8313b9aaf57ae3bf97f9c0dcbd4fd71488d775

File: aws/modules/register_group/dns.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+resource "aws_route53_record" "multi" {
+  count = "${length(var.registers)}"
```
Comment (philandstuff) [inline aws/modules/register_group/dns.tf]: what's this?

File: aws/registers/registers.tf [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+
+  name = "territory"
+  environment = "${var.vpc_name}"
+  load_balancer = "${module.multi.load_balancer}"
+  dns_zone_id = "${aws_route53_zone.temporary_hosted_zone.zone_id}"
+
+  enable_availability_checks = "${var.enable_availability_checks}"
+}
+
+module "uk_register" {
```
Comment (philandstuff) [inline aws/registers/registers.tf]: this will need an update to the README (probably after #246 is merged) to document an extra step for creating a register

File: aws/modules/register_group/dns.tf (hunk 3 above)
Comment (samcrang) [inline aws/modules/register_group/dns.tf]: you're right--this needs to go.

**Comment by philandstuff** [comment]
other than those comments ☝️, this looks good

File: aws/registers/registers.tf (hunk 4 above)
Comment (samcrang) [inline aws/registers/registers.tf]: Yup, I was just about to add a comment on your PR about this 😉

File: aws/modules/register_group/dns.tf (hunk 3 above)
Comment (samcrang) [inline aws/modules/register_group/dns.tf]: Actually this is still needed so we don't cause any downtime when we deploy these changes. This can be removed later as part of cleanup.

This PR creates a temporary hosted zone which contains the records deployed in the new way. I'll swap over the `NS` records on `openregister.org` manually to this new Hosted Zone before doing some cleanup, applying these DNS changes to the Hosted Zone that's created by the `core` Terraform module, and swapping the `NS` records back.

**Comment by samcrang** [comment]
@philandstuff how's this?
